src/Psy/Plugin/Manager.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Psy\Plugin;
+
+class Manager
+{
+    /** @var AbstractPlugin[] */
+    protected static $plugins = array();
+
+    /**
+     * @param AbstractPlugin $plugin
+     * @param $name
+     */
+    public static function register(AbstractPlugin $plugin, $name)
+    {
+        self::$plugins[$name] = $plugin;
+    }
```

Comment:
What is the reason behind that? why the plugin has to be able to modify the manager? The manager should not be accessible, if the manager goes public we might be needing some intermediate layer for protection like the registry one you said before.